go-packages/meep-logger/logger.go

```diff
@@ -17,59 +17,117 @@
+        logrus.SetLevel(logrus.DebugLevel)
 	componentName = name
 }
 
 func MeepJSONLogInit(name string) {
-	log.SetFormatter(&log.JSONFormatter{})
-	log.SetLevel(log.DebugLevel)
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	//logrus.SetLevel(logrus.TraceLevel)
+        logrus.SetLevel(logrus.DebugLevel)
 	componentName = name
 }
 
+// getLogCaller
+// setReportCaller cannot ignore caller levels, so cannot work in a wrapper. Fix not merged in github, so doing it manually
+func getLogCaller() string {
+	_, file, line, _ := runtime.Caller(2)
+
+        location := fmt.Sprintf("%v:%v", path.Base(file), line)
+	return location
+}
+
 func Info(args ...interface{}) {
-	log.WithFields(log.Fields{
+	logrus.WithFields(logrus.Fields{
 		"meep.component": componentName,
+		"meep.time": time.Now().String(),
 	}).Info(args...)
 }
 
 func Debug(args ...interface{}) {
-	log.WithFields(log.Fields{
+	logrus.WithFields(logrus.Fields{
 		"meep.component": componentName,
+		"meep.from": getLogCaller(),
+		"meep.time": time.Now().String(),
 	}).Debug(args...)
 }
 
+func Trace(args ...interface{}) {
+        logrus.WithFields(logrus.Fields{
+                "meep.component": componentName,
+                "meep.from": getLogCaller(),
+                "meep.time": time.Now().String(),
+        }).Trace(args...)
+}
+
 func Warn(args ...interface{}) {
-	log.WithFields(log.Fields{
+	logrus.WithFields(logrus.Fields{
 		"meep.component": componentName,
+		"meep.from": getLogCaller(),
+		"meep.time": time.Now().String(),
 	}).Warn(args...)
 }
+
 func Error(args ...interface{}) {
-	log.WithFields(log.Fields{
+	logrus.WithFields(logrus.Fields{
 		"meep.component": componentName,
+		"meep.from": getLogCaller(),
+		"meep.time": time.Now().String(),
 	}).Error(args...)
 }
+
 func Panic(args ...interface{}) {
-	log.WithFields(log.Fields{
+	logrus.WithFields(logrus.Fields{
 		"meep.component": componentName,
+		"meep.from": getLogCaller(),
+		"meep.time": time.Now().String(),
 	}).Panic(args...)
 }
 
 func Fatal(args ...interface{}) {
-	log.WithFields(log.Fields{
+	logrus.WithFields(logrus.Fields{
 		"meep.component": componentName,
+		"meep.from": getLogCaller(),
+		"meep.time": time.Now().String(),
 	}).Fatal(args...)
 }
 
-func WithFields(fields Fields) *log.Entry {
-	return log.WithFields(log.Fields(fields))
+func WithFields(fields Fields) *logrus.Entry {
+	return logrus.WithFields(logrus.Fields(fields))
+}
+
+func httpLog(args ...interface{}) {
+        logrus.WithFields(logrus.Fields{
+                "meep.component": componentName,
+                "meep.from": getLogCaller(),
+                "meep.time": time.Now().String(),
+        }).Debug(args...)
+}
+
+func HTTP(inner http.Handler, name string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+
+		inner.ServeHTTP(w, r)
+
+		httpLog(
+			r.Method,
+			" ", r.RequestURI,
+			" apiFct: ", name,
+			" execTime: ", time.Since(start))
+	})
```

Comment:
Why is a logging package serving HTTP? This should probably not be in this package.

Reply:
changed to have logger created from swagger to call meepLogger